### DISCUSSION TITLE
Expand System API

### DISCUSSION THREAD

**Discussion by gutierrezx7** [gameserverapp/Platform#700] [Suggestions]
**Describe the improvement**
Post and get requests to the API to collect player information such as: through Discord finding the Player's EOSID, collecting information about the number of tokens, sending tokens, reducing tokens, basic functionalities for future integrations with developments on Discord. 

> _This information is just for an example of functionality, it does not necessarily need to be this way or just this._

**Section**
API and Integrations

**Additional context**
I'm available to assist with testing and features, I'm an active developer and would love to work with tokens and my players' information on my Discord.

**Comment by mvaessen** [comment]
Hi, thanks for your suggestion. We are planning to expand the `System API`. This API is currently limitedly available.

Current docs: https://docs.gameserverapp.com/developers/system-api/

**Comment by XanaOG** [comment]
> It would be nice if GSA had a endpoint that logs the game server console (Basically a endpoint exposing the logs so it gives us a option to still get live-updates without interfering with GSA's functionality)

> Could work by passing a Game server ID as a query and when using proper authentication methods for that server it would essentially expose all content displayed in the game server console on the dashboard via a endpoint

Also possibly the ability to execute commands on the gameserver via the API

**Comment by mvaessen** [comment]
the game server endpoints are currently available: https://docs.gameserverapp.com/developers/system-api/all-game-servers

**Comment by jakubpech** [comment]
**Send deliveries via API**

If there will be an API for Player/Character operations, would it also be possible to add a "Send Delivery" call?
The use case is interconnecting external shops/bank API polling scripts with the GSA delivery system.

**Comment by mvaessen** [comment]
Thanks for the suggestions!
A lot of endpoints for Player, Character and group data were added 2 weeks ago. Later today an update will be deployed with endpoints for:
- kick
- ban/unban
- mute/unmute
- mutate tokens
- add note
- add/remove custom roles

https://docs.gameserverapp.com/developers/system-api/introduction

Right now the best way to send deliveries, is by using the webhooks. I may add support through the system api later.
